src/DynamoPackages/PackageLoader.cs

```diff
@@ -748,23 +748,32 @@ private void CheckPackageNodeLibraryCertificates(string packageDirectoryPath, Pa
         /// <param name="filename">The filename of a DLL</param>
         /// <param name="assem">out Assembly - the passed value does not matter and will only be set if loading succeeds</param>
         /// <returns>Returns Success if success, NotManagedAssembly if BadImageFormatException, AlreadyLoaded if FileLoadException</returns>
-        internal static AssemblyLoadingState TryReflectionOnlyLoadFrom(string filename, out Assembly assem)
+        internal static AssemblyLoadingState TryMetaDataContextLoad(string filename,MetadataLoadContext mlc, out Assembly assem)
         {
-            try
-            {
-                assem = Assembly.ReflectionOnlyLoadFrom(filename);
-                return AssemblyLoadingState.Success;
-            }
-            catch (BadImageFormatException)
-            {
-                assem = null;
-                return AssemblyLoadingState.NotManagedAssembly;
-            }
-            catch (FileLoadException)
-            {
-                assem = null;
-                return AssemblyLoadingState.AlreadyLoaded;
-            }
+
+                try
+                {
+                var assemName = new AssemblyName(filename);
+                //MLC will return assembly if names and mvid match
```

Comment:
mvid is a guid that is baked into each assembly on compilation, it's usually different even if the source is the same. In the case of this test we have literally copied an assembly twice so it's the same - I had to make this change to get the test to fail and keep the old behavior. I have no idea how to evaluate if this behavior is useful or not, but I figured at this point I'd rather not change it close to release.